### PR TITLE
Take screenshot

### PR DESCRIPTION
**Pull request by onmyway133**
- Remove protocol extensions for UsesCommandLine and UsesXcodebuild, and use composition instead
- Refine the configure methods
- Screenshot task needs to specify app and test schemes
- Parse xcresult folder
- Fix header search path for SPMUtility package
- Update TestPuma workspace
- Update README. Introduce Documentation folder
- Support test plan in Xcode 11

### CODE REVIEW

File: Documentation/Tasks/Screenshot.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## Take screenshot
+
+### Create UITest scheme
+
+Puma leverages UITest scheme to take screenshot. You are also encourage to use test plan feature of Xcode 11, but not required.
```
Comment (besarism) [inline Documentation/Tasks/Screenshot.md]: You are also **encouraged**

File: Documentation/Compile.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+## How to compile Puma from source
+
+Head over to [Swift Package Manager usage](https://github.com/apple/swift-package-manager/blob/master/Documentation/Usage.md) to create an executable.
+
+There's an already created [ManualCompile](https://github.com/pumaswift/Puma/tree/develop/Example/TestApp/ManualCompile) for your reference.
+
+Step 1: In your project folder, run these to create Swift Package Manager structure. Create another folder called, for example ManulCompile to keep our script.
```
Comment (thii) [inline Documentation/Compile.md]: Typo: ManualCompile

File: Sources/Core/Extensions/Folder+Extension.swift [hunk 3]
```diff
@@ -0,0 +1,22 @@
+//
+//  Folder+Extension.swift
+//  PumaCore
+//
+//  Created by khoa on 15/12/2019.
+//
```
Comment (thii) [inline Sources/Core/Extensions/Folder+Extension.swift]: You can create a template to make header comments consistent for all files.

https://oleb.net/blog/2017/07/xcode-9-text-macros/
https://useyourloaf.com/blog/changing-xcode-header-comment/
https://help.apple.com/xcode/mac/9.0/index.html?localePath=en.lproj#/dev91a7a31fc

Comment (onmyway133) [reply Sources/Core/Extensions/Folder+Extension.swift]: @thii Thanks for the tip, I tried adding in this commit https://github.com/pumaswift/Puma/pull/20/commits/a5a77f385fcb152d35a08c8bca960950851f6fe6 but it does not seem to work

File: Documentation/Tasks/Screenshot.md (hunk 1 above)
Comment (onmyway133) [reply Documentation/Tasks/Screenshot.md]: Fixed

File: Documentation/Compile.md (hunk 2 above)
Comment (onmyway133) [reply Documentation/Compile.md]: Fixed

**Comment by onmyway133** [comment]
@thii I merge this for now, let's tackle header comment in another PR

**Pull request merged**